openspp-project/openspp-registry
# Improve registry import templates

## Why is this change needed?

Need to improve the import template of Individual and Group Registry.

## How was the change implemented?

- Extends the class `ImportRecords ` to modify the function `importRecords` in javascript to add in the context the parameters `default_is_registrant` and `default_is_group`
- Extends the model `res.partner` to modify the import template based on the context provided

## New test case

```
TestRegistrant::test_08_get_import_templates
```

## How to test manually...

pre-requisites
- Install spp_registrant_import module
- Go to Registry

Test 1

- Go to Group.
- Click actions (The cog wheel beside the page name) then click Import records.
- Check that the download template button's text is now "Import Template for Groups" instead of "Import Template for Customers".
- Download the template.
- Check the the downloaded xlsx.
- Modify the xlsx or leave it as it is.
- Click the Upload File button.
- Select the downloaded xlsx.
- Click Import.
- Check if the import is success.

Test 2
- Go to Individual.
- Click actions (The cog wheel beside the page name) then click Import records.
- Check that the download template button's text is now "Import Template for Individuals" instead of "Import Template for Customers".
- Download the template.
- Check the the downloaded xlsx.
- Modify the xlsx or leave it as it is.
- Click the Upload File button.
- Select the downloaded xlsx.
- Click Import.
- Check if the import is success.

## Links
- https://github.com/OpenSPP/openspp-acf/issues/31
